irods/client_configuration/__init__.py

```diff
@@ -16,6 +16,13 @@ class iRODSConfiguration(object):
     __slots__ = ()
 
 def getter(category, setting):
+    """A programmatic way of allowing the current value of the specified setting be given
+    indirectly (through an extra call indirection) as the default value of a parameter.
+
+    Returns a callable that, in turn, will yield the setting's value.
+
+    See this function's usage in irods.manager.data_object_manager.open for an example.
```

Comment:
Is it clear to other python devs this is accessing the info returned by `globals()`?
How does the dev know where the data is coming from? Does that matter in this case?

Reply:
> Is it clear to other python devs this is accessing the info returned by `globals()`? How does the dev know where the data is coming from? Does that matter in this case?

Just a little background, `globals( )` is possibly misleading in name if you're more accustomed to the C/C++ universe.  In Python  developers will usually(?) know it  returns the dictionary for the module containing the current scope in which the call is being made.  So here, we're returning the "global" or top-level attributes of module irods.client_configuration, that's exactly what we want to access - so irods.client_configuration.getter('data_objects','auto_close') returns a getter (callable) getter for the setting-in-question's value, which will be in that "global" dict....

Were you meaning to ask for  putting a mention of this function's use of `global` as (1) a warning to other devs, or (2) in order to give more explication to the inner workings?

Reply:
> Were you meaning to ask for putting a mention of this function's use of `global` in the doc-string as (1) a warning to other devs, or (2) in order to give more explication to the inner workings?

Yes. It's not clear to me whether python devs **just** know what is happening with the function. If the use of the `getter` is common for other python devs, then it's fine.

However, if devs have to peek inside to understand what is happening, then that's a sign that we need to include a statement in the docstring about that.

Reply:
Ok... now the docstring for `getter` is a bit clearer, hopefully.... : )

Reply:
Yes. Much better.